assets/build/install.sh

```diff
@@ -62,64 +71,67 @@ GITLAB_SHELL_VERSION=${GITLAB_SHELL_VERSION:-$(cat ${GITLAB_INSTALL_DIR}/GITLAB_
 chown -R ${GITLAB_USER}: ${GITLAB_SHELL_INSTALL_DIR}
 
 cd ${GITLAB_SHELL_INSTALL_DIR}
-exec_as_git cp -a ${GITLAB_SHELL_INSTALL_DIR}/config.yml.example ${GITLAB_SHELL_INSTALL_DIR}/config.yml
+exec_as_git cp -a config.yml.example config.yml
 if [[ -x ./bin/compile ]]; then
   echo "Compiling gitlab-shell golang executables..."
-  exec_as_git PATH=/tmp/go/bin:$PATH GOROOT=/tmp/go ./bin/compile
+  ./bin/compile
+  rm -rf go_build
 fi
-exec_as_git ./bin/install
+./bin/install
 
 # remove unused repositories directory created by gitlab-shell install
-exec_as_git rm -rf ${GITLAB_HOME}/repositories
+rm -rf ${GITLAB_HOME}/repositories
 
 # download gitlab-workhorse
 echo "Cloning gitlab-workhorse v.${GITLAB_WORKHORSE_VERSION}..."
-exec_as_git git clone -q -b v${GITLAB_WORKHORSE_VERSION} --depth 1 ${GITLAB_WORKHORSE_URL} ${GITLAB_WORKHORSE_INSTALL_DIR}
-chown -R ${GITLAB_USER}: ${GITLAB_WORKHORSE_INSTALL_DIR}
+git clone -q -b v${GITLAB_WORKHORSE_VERSION} --depth 1 ${GITLAB_WORKHORSE_URL} ${GITLAB_WORKHORSE_BUILD_DIR}
+make -C ${GITLAB_WORKHORSE_BUILD_DIR} install
 
-#install gitlab-workhorse
-cd ${GITLAB_WORKHORSE_INSTALL_DIR}
-PATH=/tmp/go/bin:$PATH GOROOT=/tmp/go make install
+# clean up
+rm -rf ${GITLAB_WORKHORSE_BUILD_DIR}
 
-#download pages
+# download gitlab-pages
 echo "Downloading gitlab-pages v.${GITLAB_PAGES_VERSION}..."
-exec_as_git git clone -q -b v${GITLAB_PAGES_VERSION} --depth 1 ${GITLAB_PAGES_URL} ${GITLAB_PAGES_INSTALL_DIR}
-chown -R ${GITLAB_USER}: ${GITLAB_PAGES_INSTALL_DIR}
+git clone -q -b v${GITLAB_PAGES_VERSION} --depth 1 ${GITLAB_PAGES_URL} ${GITLAB_PAGES_BUILD_DIR}
+
+# install gitlab-pages
+make -C ${GITLAB_PAGES_BUILD_DIR}
+cp -a ${GITLAB_PAGES_BUILD_DIR}/gitlab-pages /usr/local/bin/
 
-#install gitlab-pages
-cd ${GITLAB_PAGES_INSTALL_DIR}
-PATH=/tmp/go/bin:$PATH GOROOT=/tmp/go make
-cp -f gitlab-pages /usr/local/bin/
+# clean up
+rm -rf ${GITLAB_PAGES_BUILD_DIR}
 
-# download gitaly
+# download and build gitaly
 echo "Downloading gitaly v.${GITALY_SERVER_VERSION}..."
-exec_as_git git clone -q -b v${GITALY_SERVER_VERSION} --depth 1 ${GITLAB_GITALY_URL} ${GITLAB_GITALY_INSTALL_DIR}
-chown -R ${GITLAB_USER}: ${GITLAB_GITALY_INSTALL_DIR}
-# copy default config for gitaly
-exec_as_git cp ${GITLAB_GITALY_INSTALL_DIR}/config.toml.example ${GITLAB_GITALY_INSTALL_DIR}/config.toml
+git clone -q -b v${GITALY_SERVER_VERSION} --depth 1 ${GITLAB_GITALY_URL} ${GITLAB_GITALY_BUILD_DIR}
 
 # install gitaly
-cd ${GITLAB_GITALY_INSTALL_DIR}
-ln -sf /tmp/go /usr/local/go
-PATH=/tmp/go/bin:$PATH make install && make clean
-rm -f /usr/local/go
+make -C ${GITLAB_GITALY_BUILD_DIR} install
+mkdir -p ${GITLAB_GITALY_INSTALL_DIR}
+cp -a ${GITLAB_GITALY_BUILD_DIR}/ruby ${GITLAB_GITALY_INSTALL_DIR}/
+cp -a ${GITLAB_GITALY_BUILD_DIR}/config.toml.example ${GITLAB_GITALY_INSTALL_DIR}/config.toml
+rm -rf ${GITLAB_GITALY_INSTALL_DIR}/ruby/vendor/bundle/ruby/**/cache
+chown -R ${GITLAB_USER}: ${GITLAB_GITALY_INSTALL_DIR}
```

Comment:
I've altered the installation of gitaly bit in the interest of reducing the size of the image.
Now we build gitaly in a temporary build directory (`GITLAB_GITALY_BUILD_DIR`) and copy over just the required artifacts to the `GITLAB_GITALY_INSTALL_DIR`)